File: pages/dashboard/profile-settings.js
// 人物分析 · 分析设置 tab（v0.4.0）：19 项 profile_* 配置分组表单
// 字段与 PRD §5 一一对应；备用模型复用 v0.3.2 拖拽排序弹窗组件（summary-settings.js 导出）；
// 「图片渲染」分组仅说明文字 —— 与消息总结共用 summary_t2i_* 配置，不重复表单
// 端点契约（PRD §3.8）：GET/POST profile/settings、POST profile/settings/reset、GET profile/providers
import { bridge, showToast, el } from "./common.js";
import {
    openProviderMultiModalEx,
    refreshProviderMultiTrigger,
    parseProviderMultiValue,
} from "./summary-settings.js";

// 19 项人物分析配置元数据：分组 / 展示名 / 说明 / 控件类型
// kind 与后端 PROFILE_TYPES（bool/int/str/list）对应，请求封装沿用 bridge.apiGet/apiPost
const PROFILE_FIELD_META = [
    // —— 基础 ——
    { key: "profile_enabled", group: "基础", label: "功能总开关", desc: "关闭后 /人物分析 指令与 Web 发起分析均不可用", kind: "bool" },
    { key: "profile_permission", group: "基础", label: "指令权限", desc: "人物画像涉及个体隐私，默认仅管理员可在群里触发", kind: "select", options: [["admin", "仅管理员（admin）"], ["all", "所有人（all）"]] },
    { key: "profile_output_mode", group: "基础", label: "输出模式", desc: "指令触发时的输出形态；Web 发起分析始终在页面内渲染结果", kind: "select", options: [["forward", "合并转发（forward）"], ["image", "文转图（image）"], ["text", "纯文本（text）"]] },
    // —— 模型 ——
    { key: "profile_provider", group: "模型", label: "主选 LLM 提供商", desc: "人物分析独立一套模型配置，与消息总结互不干扰；留空时回退当前会话 provider", kind: "provider" },
    { key: "profile_fallback_providers", group: "模型", label: "备用模型列表", desc: "主选模型失败后按勾选顺序逐个尝试，全部失败回退会话模型", kind: "provider_multi" },
    // —— 参数上限 ——
    { key: "profile_max_count", group: "参数上限", label: "最大分析条数", desc: "单次分析最多纳入的目标消息条数（分页拉取上限）", kind: "int", unit: "条", min: 1 },
    { key: "profile_max_prompt_chars", group: "参数上限", label: "素材长度预算", desc: "送入 LLM 的素材字符上限，超出保留最近消息截断（统计仍全量）", kind: "int", unit: "字符", min: 1000 },
    // —— 关系上下文 ——
    { key: "profile_relation_context", group: "关系上下文", label: "关系上下文开关", desc: "开启后双向识别 @/回复 互动对象并拉取双方消息，用于人物关系分析；任一环节失败自动降级，不阻断主分析", kind: "bool" },
    { key: "profile_relation_max_partners", group: "关系上下文", label: "最大互动对象数", desc: "互动对象排行取 Top N，其消息作为关系分析的双方对话上下文", kind: "int", unit: "人", min: 1 },
    // —— 分析维度（5 项多选） ——
    { key: "profile_dim_habits", group: "分析维度", label: "发言习惯", desc: "频率 / 平均长度 / 口头禅与高频表达 / 语气与表情使用 / 活跃时段解读", kind: "bool" },
    { key: "profile_dim_activity", group: "分析维度", label: "活动时间规律", desc: "按小时 + 按星期发言分布图表与简短解读", kind: "bool" },
    { key: "profile_dim_personality", group: "分析维度", label: "性格分析", desc: "基于发言内容的性格推断（外向/理性/幽默等，需给出依据）", kind: "bool" },
    { key: "profile_dim_hobbies", group: "分析维度", label: "兴趣爱好", desc: "从发言主题归纳兴趣领域", kind: "bool" },
    { key: "profile_dim_relations", group: "分析维度", label: "人物关系", desc: "与 Top 互动对象的关系刻画、互动模式推测", kind: "bool" },
    // —— 限流 ——
    { key: "profile_user_cooldown", group: "限流", label: "用户触发冷却", desc: "同一用户两次触发人物分析的最小间隔", kind: "int", unit: "秒", min: 0 },
    { key: "profile_group_cooldown", group: "限流", label: "群触发冷却", desc: "同一群两次触发人物分析的最小间隔", kind: "int", unit: "秒", min: 0 },
    // —— 触发反馈 ——
    { key: "profile_feedback_mode", group: "触发反馈", label: "触发反馈模式", desc: "指令触发后的即时反馈；reaction=贴表情回应（协议端不支持自动降级文字）", kind: "select", options: [["reaction", "贴表情回应（reaction）"], ["text", "文字提示（text）"], ["none", "关闭（none）"]] },
    { key: "profile_feedback_text", group: "触发反馈", label: "触发反馈文案", desc: "text 模式的提示文案；reaction 失败降级时同用；留空回退内置默认", kind: "text" },
    // —— 存储保留 ——
    { key: "profile_keep_days", group: "存储保留", label: "历史保留天数", desc: "历史分析 JSON 保留天数，每日定时清理过期文件", kind: "int", unit: "天", min: 1 },
];

let profileProviders = []; // LLM 提供商列表（可能为空数组 → 下拉仅含回退项）

async function loadProfileSettings() {
    const container = document.getElementById("profileSettingsForm");
    try {
        // 设置与 providers 并行拉取；providers 失败不阻塞表单渲染（降级为空列表）
        const [data, provData] = await Promise.all([
            bridge.apiGet("profile/settings"),
            bridge.apiGet("profile/providers").catch(() => ({ providers: [] })),
        ]);
        profileProviders = provData.providers || [];
        renderProfileForm(data.settings || {});
    } catch (e) {
        container.innerHTML = "";
        container.appendChild(el("div", "loading", "人物分析设置加载失败"));
        showToast("加载人物分析设置失败: " + (e.message || "未知错误"), "error");
    }
}

function renderProfileForm(settings) {
    const container = document.getElementById("profileSettingsForm");
    container.innerHTML = "";
    let currentGroup = null;
    let groupEl = null;
    for (const meta of PROFILE_FIELD_META) {
        if (meta.group !== currentGroup) {
            currentGroup = meta.group;
            groupEl = el("div", "summary-group");
            groupEl.appendChild(el("div", "summary-group-title", meta.group));
            container.appendChild(groupEl);
        }
        groupEl.appendChild(buildProfileItem(meta, settings[meta.key] ?? ""));
    }
    // 「图片渲染」分组：与消息总结共用 summary_t2i_* 配置，仅说明 + 跳转入口
    container.appendChild(buildT2iNoteGroup());
}

// 图片渲染说明分组（PRD 3.6：渲染基础设施配置共享，不新增 profile_t2i_ 冗余项）
function buildT2iNoteGroup() {
    const groupEl = el("div", "summary-group");
    groupEl.appendChild(el("div", "summary-group-title", "图片渲染"));
    const note = el("div", "profile-note");
    note.appendChild(
        el(
            "div",
            "profile-note-text",
            "人物报告的图片渲染（主题模式 / 深浅色时段 / 渲染超时 / CDN 节点顺序）与「消息总结」共用同一套配置（summary_t2i_*），不在此重复列出。调整请前往：消息总结 → 总结设置 → 图片渲染。",
        ),
    );
    const goBtn = el("button", "btn btn-ghost btn-sm", "前往「总结设置」调整 ›");
    goBtn.type = "button";
    goBtn.addEventListener("click", () => {
        // 分区切换后 switchScope 自动激活首个子 tab（总结设置）
        const scopeBtn = document.querySelector('.scope-btn[data-scope="summary"]');
        if (scopeBtn) scopeBtn.click();
    });
    note.appendChild(goBtn);
    groupEl.appendChild(note);
    return groupEl;
}

function buildProfileItem(meta, rawValue) {
    const stacked = meta.kind === "provider_multi";
    const item = el("div", "s-item" + (stacked ? " s-item-stack" : ""));
    const info = el("div", "setting-info");
    info.appendChild(el("div", "setting-name", meta.label));
    info.appendChild(el("div", "setting-desc", meta.desc));
    item.appendChild(info);

    const control = el("div", "setting-control");
    switch (meta.kind) {
        case "bool": {
            const label = el("label", "all-mode-switch");
            const input = document.createElement("input");
            input.type = "checkbox";
            input.dataset.key = meta.key;
            input.checked = String(rawValue).toLowerCase() === "true";
            const track = el("span", "switch-track");
            track.appendChild(el("span", "switch-thumb"));
            label.appendChild(input);
            label.appendChild(track);
            control.appendChild(label);
            break;
        }
        case "int": {
            const input = document.createElement("input");
            input.type = "number";
            input.className = "input input-sm";
            input.min = String(meta.min ?? 0);
            input.step = "1";
            input.dataset.key = meta.key;
            input.value = String(rawValue);
            control.appendChild(input);
            if (meta.unit) control.appendChild(el("span", "unit", meta.unit));
            break;
        }
        case "select": {
            const select = document.createElement("select");
            select.className = "input input-select";
            select.dataset.key = meta.key;
            for (const [val, text] of meta.options) {
                const opt = el("option", null, text);
                opt.value = val;
                select.appendChild(opt);
            }
            select.value = String(rawValue);
            control.appendChild(select);
            break;
        }
        case "provider": {
            const select = document.createElement("select");
            select.className = "input input-select provider-select";
            select.dataset.key = meta.key;
            // 首项固定为空值「回退会话 provider」；providers 为空数组时下拉仅含此项
            const fallback = el("option", null, "（回退会话 provider）");
            fallback.value = "";
            select.appendChild(fallback);
            for (const prov of profileProviders) {
                const opt = el("option", null, prov.name || prov.id);
                opt.value = prov.id;
                select.appendChild(opt);
            }
            // 当前值不在列表中（provider 已删除）→ 以禁用项保留原值，避免静默清空
            if (rawValue && ![...select.options].some((o) => o.value === String(rawValue))) {
                const ghost = el("option", null, `${rawValue}（已失效）`);
                ghost.value = String(rawValue);
                ghost.disabled = true;
                select.appendChild(ghost);
            }
            select.value = String(rawValue ?? "");
            control.appendChild(select);
            break;
        }
        case "provider_multi": {
            // 复用 v0.3.2 拖拽排序弹窗组件：触发按钮 + 隐藏 input（JSON 字符串数组）
            control.className = "setting-control";
            const selected = parseProviderMultiValue(rawValue);
            const hidden = document.createElement("input");
            hidden.type = "hidden";
            hidden.dataset.key = meta.key;
            hidden.value = JSON.stringify(selected);
            control.appendChild(hidden);
            const trigger = el("button", "btn btn-ghost provider-multi-trigger");
            trigger.type = "button";
            control.appendChild(trigger);
            refreshProviderMultiTrigger(trigger, selected, profileProviders);
            trigger.addEventListener("click", () =>
                openProviderMultiModalEx({
                    formSelector: "#profileSettingsForm",
                    key: meta.key,
                    trigger,
                    providers: profileProviders,
                    title: "🔀 备用人物分析模型 · 降级顺序",
                }),
            );
            break;
        }
        case "text": {
            const input = document.createElement("input");
            input.type = "text";
            input.className = "input input-text";
            input.dataset.key = meta.key;
            input.value = String(rawValue ?? "");
            control.appendChild(input);
            break;
        }
    }
    item.appendChild(control);
    return item;
}

// 用接口返回的全量 settings 回填表单（保存/重置成功后复用）
function fillProfileForm(settings) {
    for (const meta of PROFILE_FIELD_META) {
        if (!(meta.key in settings)) continue;
        const ctrl = document.querySelector(`#profileSettingsForm [data-key="${meta.key}"]`);
        if (!ctrl) continue;
        const raw = settings[meta.key];
        if (meta.kind === "bool") ctrl.checked = String(raw).toLowerCase() === "true";
        else if (meta.kind === "provider_multi") {
            // ctrl 为隐藏 input[data-key]，写入 JSON 字符串数组并刷新同列触发按钮摘要
            const ids = parseProviderMultiValue(raw);
            ctrl.value = JSON.stringify(ids);
            const trig = ctrl.parentElement?.querySelector(".provider-multi-trigger");
            if (trig) refreshProviderMultiTrigger(trig, ids, profileProviders);
        } else ctrl.value = String(raw ?? "");
    }
}

// 收集全部 19 项组装 save 载荷；前端基础校验失败返回 null（后端 400 为最终裁决）
function collectProfileSettings() {
    const settings = {};
    for (const meta of PROFILE_FIELD_META) {
        const ctrl = document.querySelector(`#profileSettingsForm [data-key="${meta.key}"]`);
        if (!ctrl) continue;
        switch (meta.kind) {
            case "bool":
                settings[meta.key] = ctrl.checked;
                break;
            case "int": {
                const raw = ctrl.value.trim();
                if (!/^\d+$/.test(raw)) {
                    showToast(`「${meta.label}」需为非负整数`, "error");
                    return null;
                }
                const v = parseInt(raw, 10);
                if (v < (meta.min ?? 0)) {
                    showToast(`「${meta.label}」不能小于 ${meta.min}`, "error");
                    return null;
                }
                settings[meta.key] = v;
                break;
            }
            case "provider_multi":
                // 值存于隐藏 input（JSON 字符串数组，顺序即降级顺序，含 stale 失效项）
                settings[meta.key] = parseProviderMultiValue(ctrl.value);
                break;
            default:
                settings[meta.key] = ctrl.value;
        }
    }
    return settings;
}

async function saveProfileSettings() {
    const settings = collectProfileSettings();
    if (!settings) return;
    const btn = document.getElementById("saveProfileBtn");
    btn.disabled = true;
    try {
        // PRD §3.8：保存为 POST /profile/settings（与 summary 的 settings/save 不同，严格按端点契约）
        const data = await bridge.apiPost("profile/settings", { settings });
        fillProfileForm(data.settings || settings);
        showToast("人物分析设置已保存", "success");
    } catch (e) {
        showToast("保存失败: " + (e.message || "未知错误"), "error");
    } finally {
        btn.disabled = false;
    }
}

async function resetProfileAll() {
    if (!confirm(`确定将全部 ${PROFILE_FIELD_META.length} 项人物分析设置恢复为默认值吗？此操作不可撤销。`)) return;
    try {
        const data = await bridge.apiPost("profile/settings/reset", {});
        fillProfileForm(data.settings || {});
        showToast("已恢复全部默认设置", "success");
    } catch (e) {
        showToast("重置失败: " + (e.message || "未知错误"), "error");
    }
}

// ========== 事件绑定：人物分析设置 ==========
function bindProfileSettingsEvents() {
    document.getElementById("saveProfileBtn").addEventListener("click", saveProfileSettings);
    document.getElementById("resetProfileAllBtn").addEventListener("click", resetProfileAll);
    // 备用模型弹窗的 取消/完成/防误触 事件已由 bindSummarySettingsEvents 统一绑定（弹窗 DOM 全局唯一）
}

export { loadProfileSettings, bindProfileSettingsEvents };

File: core/db_config/profile_settings.py
"""人物分析功能配置 Mixin（ProfileSettingsMixin）。

人物分析功能 19 项配置（PROFILE_DEFAULTS / PROFILE_TYPES）的读写与全量校验。
类常量随 Mixin 迁移，组装后经 MRO 仍可从 ConfigManager 直接访问
（web_api.py 直接引用 ConfigManager.PROFILE_*）。
"""

import json
from typing import Any

from astrbot.api import logger


class ProfileSettingsMixin:
    """人物分析功能配置 Mixin：19 项配置（PROFILE_DEFAULTS/TYPES）+ 批量保存校验。"""

    # ========== 人物分析功能配置常量（v0.4.0） ==========

    # 人物分析功能 19 项配置的默认值（见 PRD §5）。值一律字符串存储；列表类型 JSON 序列化。
    # 同时作为初始化播种目标与 dashboard「恢复默认」的目标值；default 一律非 null。
    PROFILE_DEFAULTS: dict[str, str] = {
        # 人物分析总开关
        "profile_enabled": "true",
        # 指令权限：admin 仅管理员 / all 所有人
        "profile_permission": "admin",
        # 输出模式：forward 合并转发 / image 图片 / text 纯文本
        "profile_output_mode": "forward",
        # 主选 LLM provider id（空=使用当前会话模型）
        "profile_provider": "",
        # 备用 provider 降级列表（JSON 存储）；主选失败后按序尝试，全失败兜底会话模型
        "profile_fallback_providers": "[]",
        # 单次分析最大消息条数
        "profile_max_count": "2000",
        # 送入 LLM 的素材长度预算（字符数，截断保最近、统计仍全量）
        "profile_max_prompt_chars": "60000",
        # 关系上下文开关：开启后双向识别目标↔他人的 @/回复互动对象
        "profile_relation_context": "true",
        # 互动对象上下文最大人数（Top N）
        "profile_relation_max_partners": "10",
        # 分析维度开关（五项，关闭的维度不进 prompt、不渲染）
        "profile_dim_habits": "true",
        "profile_dim_activity": "true",
        "profile_dim_personality": "true",
        "profile_dim_hobbies": "true",
        "profile_dim_relations": "true",
        # 触发冷却（秒）：用户 / 群双维度限流
        "profile_user_cooldown": "60",
        "profile_group_cooldown": "30",
        # 触发反馈模式：reaction 贴表情 / text 文字提示 / none 关闭
        "profile_feedback_mode": "reaction",
        # 文字反馈文案（reaction 失败降级时同用）
        "profile_feedback_text": "正在生成人物画像，请稍候…",
        # 历史分析保留天数（清理调度用）
        "profile_keep_days": "30",
    }

    # 各配置键的值类型声明（字符串形式 "bool"/"int"/"str"/"list"），供
    # get_profile_setting_typed() 类型化读取与 save_profile_settings() 全量校验使用
    PROFILE_TYPES: dict[str, str] = {
        "profile_enabled": "bool",
        "profile_permission": "str",
        "profile_output_mode": "str",
        "profile_provider": "str",
        "profile_fallback_providers": "list",
        "profile_max_count": "int",
        "profile_max_prompt_chars": "int",
        "profile_relation_context": "bool",
        "profile_relation_max_partners": "int",
        "profile_dim_habits": "bool",
        "profile_dim_activity": "bool",
        "profile_dim_personality": "bool",
        "profile_dim_hobbies": "bool",
        "profile_dim_relations": "bool",
        "profile_user_cooldown": "int",
        "profile_group_cooldown": "int",
        "profile_feedback_mode": "str",
        "profile_feedback_text": "str",
        "profile_keep_days": "int",
    }

    # ========== 人物分析配置（v0.4.0） ==========

    @staticmethod
    def _convert_profile_value(value: str, target: str):
        """将配置字符串按声明类型字符串转换，失败时抛出异常由调用方兜底。

        Args:
            value: 数据库中存储的原始字符串
            target: PROFILE_TYPES 声明的目标类型字符串（"bool"/"int"/"str"/"list"）

        Returns:
            转换后的值

        Raises:
            ValueError: 值无法解析为目标类型（如 bool 串非法、JSON 不是列表）
            json.JSONDecodeError: list 类型 JSON 解析失败
        """
        if target == "bool":
            # 不能直接 bool(value)（"false" 也是真值），必须显式匹配字符串
            lowered = value.strip().lower()
            if lowered in ("true", "1"):
                return True
            if lowered in ("false", "0"):
                return False
            raise ValueError(f"无法识别的布尔值: {value!r}")
        if target == "int":
            return int(value)
        if target == "list":
            parsed = json.loads(value)
            if not isinstance(parsed, list):
                raise ValueError(f"JSON 值不是列表: {value!r}")
            return parsed
        return value

    async def get_profile_setting(self, key: str) -> str:
        """获取人物分析配置值（字符串形式）。

        键在表中缺失时自动以规范默认值播种（INSERT OR IGNORE，绝不覆盖并发/已存值）；
        键不在 PROFILE_DEFAULTS 中时返回空串。

        Args:
            key: 配置键（PROFILE_DEFAULTS 19 项之一）

        Returns:
            str: 配置值（字符串形式）
        """
        fallback = self.PROFILE_DEFAULTS.get(key, "")
        try:
            await self._ensure_db()
            async with self.db.execute(
                "SELECT value FROM profile_settings WHERE key = ?", (key,)
            ) as cursor:
                row = await cursor.fetchone()
            if row is not None:
                return row[0]
            # 缺失键自动播种：OR IGNORE 保证并发与重复调用下不覆盖已插入值
            if key in self.PROFILE_DEFAULTS:
                await self.db.execute(
                    "INSERT OR IGNORE INTO profile_settings (key, value) VALUES (?, ?)",
                    (key, fallback),
                )
                await self.db.commit()
            return fallback
        except Exception as e:
            logger.error(f"[Profile] 获取人物分析配置失败: {e}")
            return fallback

    async def get_profile_setting_typed(self, key: str) -> Any:
        """获取人物分析配置并按 PROFILE_TYPES 声明的类型转换。

        bool 识别 "true"/"false"/"1"/"0"（大小写不敏感）；list 用 json.loads 解析。
        转换失败时记录 warning 并回退默认值的同类型结果。

        Args:
            key: 配置键

        Returns:
            声明类型的配置值（未在 PROFILE_TYPES 中声明的键按字符串原样返回）
        """
        raw = await self.get_profile_setting(key)
        target = self.PROFILE_TYPES.get(key, "str")
        try:
            return self._convert_profile_value(raw, target)
        except Exception as e:
            logger.warning(
                f"[Profile] 配置 {key} 的值 {raw!r} 类型转换失败，回退默认值: {e}"
            )
            default_raw = self.PROFILE_DEFAULTS.get(key, "")
            try:
                return self._convert_profile_value(default_raw, target)
            except Exception:
                # 默认值构造上保证合法，此分支仅为最终兜底
                return default_raw

    async def get_all_profile_settings(self) -> dict[str, Any]:
        """获取全部人物分析配置（完整 19 项，缺失以默认值补全，逐项类型化）。

        各值按 PROFILE_TYPES 声明类型转换；单项非法值回退该项默认值的同类型结果。

        Returns:
            dict[str, Any]: 按 PROFILE_DEFAULTS 顺序的完整 typed 配置字典（失败时返回空字典）
        """
        try:
            await self._ensure_db()
            async with self.db.execute(
                "SELECT key, value FROM profile_settings"
            ) as cursor:
                rows = await cursor.fetchall()
            stored = {row[0]: row[1] for row in rows}
            # 以 PROFILE_DEFAULTS 为骨架补全缺失项，保证返回恒为完整 19 项
            result: dict[str, Any] = {}
            for key, default_raw in self.PROFILE_DEFAULTS.items():
                raw = stored.get(key, default_raw)
                target = self.PROFILE_TYPES.get(key, "str")
                try:
                    result[key] = self._convert_profile_value(raw, target)
                except Exception:
                    # 非法值回退默认值的同类型结果（默认值构造上保证合法）
                    try:
                        result[key] = self._convert_profile_value(default_raw, target)
                    except Exception:
                        result[key] = default_raw
            return result
        except Exception as e:
            logger.error(f"[Profile] 获取所有人物分析配置失败: {e}")
            return {}

    async def save_profile_settings(self, settings: dict[str, Any]) -> bool:
        """批量保存人物分析配置。

        先全量校验后写入：所有传入键值全部通过 PROFILE_TYPES 声明类型校验后才
        逐项写入，任一非法则整体不写入（校验范式同 web_api summary save）。
        值归一化：bool→"true"/"false"；list/dict→JSON 序列化；其余→str()。
        未知键仅告警跳过，不阻断其余合法项的校验与写入。

        Args:
            settings: 配置键 → 值（任意可归一化类型）的字典

        Returns:
            bool: 是否保存成功（校验未通过或数据库写入异常时返回 False）
        """
        # ① 未知键检查 + ② 值归一化为存储字符串
        normalized: dict[str, str] = {}
        for key, value in settings.items():
            if key not in self.PROFILE_TYPES:
                logger.warning(f"[Profile] 忽略未知配置键: {key}")
                continue
            if isinstance(value, bool):
                str_value = "true" if value else "false"
            elif isinstance(value, (list, dict)):
                str_value = json.dumps(value, ensure_ascii=False)
            else:
                str_value = str(value)
            normalized[key] = str_value

        # ③ 逐个按声明类型校验，任一失败整体放弃写入
        for key, str_value in normalized.items():
            try:
                self._convert_profile_value(str_value, self.PROFILE_TYPES[key])
            except (ValueError, TypeError) as e:
                logger.warning(
                    f"[Profile] 配置项 {key} 的值 {str_value!r} 非法，本次整体不写入: {e}"
                )
                return False

        # ④ 全部校验通过后批量写入
        try:
            await self._ensure_db()
            for key, str_value in normalized.items():
                await self.db.execute(
                    "INSERT OR REPLACE INTO profile_settings (key, value) VALUES (?, ?)",
                    (key, str_value),
                )
            await self.db.commit()
            return True
        except Exception as e:
            logger.error(f"[Profile] 保存人物分析配置失败: {e}")
            return False

    async def reset_profile_settings(self) -> None:
        """将人物分析功能全部 19 项配置重置为 PROFILE_DEFAULTS 默认值。"""
        try:
            await self._ensure_db()
            for key, value in self.PROFILE_DEFAULTS.items():
                await self.db.execute(
                    "INSERT OR REPLACE INTO profile_settings (key, value) VALUES (?, ?)",
                    (key, value),
                )
            await self.db.commit()
        except Exception as e:
            logger.error(f"[Profile] 重置人物分析配置失败: {e}")
